### PR TITLE
Update extract dialog

### PR DESCRIPTION
**Pull request by sithlord48**
Requires: https://github.com/berv-uni-project/Frog-Archiver/pull/30

Create a proper QDialog for the extract dialog, minor clean up

### CODE REVIEW

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -42,8 +40,9 @@ void MainWindow::on_actionCompress_triggered()
 
 void MainWindow::Extract()
 {
-    extractWindow->show();
-    this->hide();
+    ExtractWindow extractWindow(new ExtractWindow(this));
```
Comment (berviantoleo) [inline src/mainwindow.cpp]: I'm not sure this is true for initialization, I think this will caused memory leak.

File: src/extractwindow.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 #include <QFileDialog>
 
 ExtractWindow::ExtractWindow(QWidget *parent) :
-    QMainWindow(parent),
+    QDialog(parent),
```
Comment (berviantoleo) [inline src/extractwindow.cpp:7]: I don't know why changes to dialog. I need more research about this.

Comment (sithlord48) [reply src/extractwindow.cpp:7]: Explained on the other CompressWIndow diff

**Review by berviantoleo — CHANGES_REQUESTED** [review]
Overall ok, but need some changes to make sure not caused memory leak, and need some description why changes to dialog.

File: src/mainwindow.cpp (hunk 1 above)
Comment (sithlord48) [reply src/mainwindow.cpp]: Yes that is a mistake.

**Review by berviantoleo — APPROVED** [review]

**Comment by sithlord48** [comment]
In the latest revision we create the dialog and reuse it. Each time it is shownif the decompression is not running the gui is reset.

**Comment by berviantoleo** [comment]
I will merge this when #33 already merged. Same with that, I just want to know more about why we not dealloc the initialization.

**Comment by berviantoleo** [comment]
Because #34 & #33 changes the pattern and do same changes in same row, this is also need to rebase.

**Review by berviantoleo — CHANGES_REQUESTED** [review]
Need to rebase

**Comment by sithlord48** [comment]
Rebased to develop

**Comment by sithlord48** [comment]
> I will merge this when #33 already merged. Same with that, I just want to know more about why we not dealloc the initialization.

We do not need to delete the CompressWindow or the ExtractWindow as they are children of the MainWindow. They will be destroyed when their parent is.  (I think this is what your asking)

**Review by berviantoleo — APPROVED** [review]

**Pull request merged**